aws-spi-pekko-http/src/main/scala/org/apache/pekko/stream/connectors/awsspi/PekkoHttpClient.scala

```diff
@@ -53,9 +60,11 @@ class PekkoHttpClient(shutdownHandle: () => Unit, private[awsspi] val connection
   lazy val runner = new RequestRunner()
 
   override def execute(request: AsyncExecuteRequest): CompletableFuture[Void] = {
-    val pekkoHttpRequest = toPekkoRequest(request.request(), request.requestContentPublisher())
```

Comment:
moving `pekkoHttpRequest` inside `runner.run` and changes below in `RequestRunner.scala` were needed to make test `naughtyHeaderCharactersDoNotGetToServer` pass.

The test is defined as:
```java
@Test
public void naughtyHeaderCharactersDoNotGetToServer() {
    String naughtyHeader = "foo\r\nbar";
    assertThatThrownBy(() -> HttpTestUtils.sendRequest(client,
                                                       SdkHttpFullRequest.builder()
                                                                         .uri(URI.create("https://localhost:" + server.port()))
                                                                         .method(SdkHttpMethod.POST)
                                                                         .appendHeader("h", naughtyHeader)
                                                                         .build())
                                          .join())
        .hasCauseInstanceOf(Exception.class);
}
```

and it was failing because `toPekkoRequest` is throwing an exception, but that exception should be inside a Future.failed. 

See a734b17e3262a8d3d27a49fef7a720d4a7ee4fc4